hadoop-common-project/hadoop-common/src/main/native/native.vcxproj

```diff
@@ -39,12 +37,14 @@
     <UseDebugLibraries>false</UseDebugLibraries>
     <WholeProgramOptimization>true</WholeProgramOptimization>
     <CharacterSet>Unicode</CharacterSet>
+    <PlatformToolset>v142</PlatformToolset>
```

Comment:
What was the old number and where was it defined?

Reply:
I think the old number should've been `v100`. I'm not sure why it wasn't defined. Visual Studio automatically added this when I ran the command to retarget the project files.

Reply:
Can we check if it was inherited from somewhere else?
This may trigger inconsistencies.

Reply:
@goiri I checked on the current trunk. There's no `PlatformToolset` or `v100` defined anywhere -

![image](https://user-images.githubusercontent.com/10280768/181075107-e72ea1bf-aed7-422e-b06b-900fef060f93.png)

![image](https://user-images.githubusercontent.com/10280768/181075278-91487b80-a183-4f3f-b0e2-156b644d3ce6.png)

Reply:
So how does it know which VS version to be scoped to?

Reply:
I ran the retarget command from `VS2019 command prompt`, thus the retarget tool added the PlatformToolset version corresponding to VS2019.
The retarget command would take care of inserting the right version based on Visual Studio command prompt that we choose. Please also note that the subsequent `mvn install` command must be run from the same command prompt.

Reply:
Given that we didn't have the target before; why would we add it now?

Reply:
> why would we add it now?

The compilation fails without the `PlatformToolset` when compiled with Visual Studio 2019 (I'm not sure about VS2010). I'm fine with documenting running the retarget command as a pre-requisite, if we don't want to add the `PlatformToolset` here.

Reply:
Take a look at HADOOP-14667
If this makes sense, we go with this.

Reply:
This is amazing. Thanks for sharing this @goiri 😊. I'm able to confirm that while building the `hadoop-common-project` module, the vcxproj and solution files automatically get upgraded to the appropriate Visual Studio version. All thanks to https://github.com/apache/hadoop/blob/95a85875d0038a73f7e6c2cdb6efdc5132d63beb/hadoop-common-project/hadoop-common/pom.xml#L844-L923

```
[INFO] --- exec-maven-plugin:1.3.1:exec (convert-ms-native-dll) @ hadoop-common ---
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\devenv.com
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\devenv.exe

Microsoft Visual Studio 2019 Version 16.11.9.
Copyright (C) Microsoft Corp. All rights reserved.
Upgrading project 'native'...
        Configuration 'Release|Win32': changing Platform Toolset to 'v142' (was 'v100').
        Configuration 'Release|x64': changing Platform Toolset to 'v142' (was 'v100').

Migration completed successfully, but some warnings were detected during migration.
For more information, see the migration report:
E:\hadoop-common-project\hadoop-common\src\main\native\UpgradeLog.htm
```

I'm able to fully build the trunk as well -
![image](https://user-images.githubusercontent.com/10280768/181500592-865ab6f6-aa1e-478a-b2cd-7e456d4f964a.png)

I'll abandon this PR since I'm able to build the trunk on Windows without these changes.